### DISCUSSION TITLE
Limit the number of resources generated for a patient.

### DISCUSSION THREAD

**Discussion by HadaRoshan** [synthetichealth/synthea#1410] [Q&A]
Hi team!
Is there any possibility in Synthea where I can limit the number of resources for a patient. For example if we want only 50 encounters for a patient, can we specify it somewhere and get only 50 encounters for that patient? Is it possible through Synthea? If yes then how?
also if it's not possible then how are you deciding the number of resources generated for a patient, Can we change that particular part of the code for our requirement? 
Can anybody help me on this one?
Thank you.

**Comment by vineetsaurabh** [comment]
I am just a beginner in Synthea. I generated patients without any encounters using a new module [Examplitis](https://github.com/synthetichealth/synthea/blob/b24977246a885dbcbc43a285c5499345feef99b8/src/test/resources/generic/example_module.json). In this module, I modified **Age_Guard** as the below:

`"allow": {
        "condition_type": "Age",
        "operator": ">",
        "quantity": 120,
        "unit": "years"
      },`

**Reply by jawalonoski** [reply]
There aren't going to be a whole lot of people over 120 years old, so I'm not sure what you're trying to accomplish.

**Reply by vineetsaurabh** [reply]
That is why the generated patients will have no encounter against this particular module, and hence the number of resources generated for a patient.

**Reply by felixwiemuth** [reply]
I just tried the same, but even when removing all other modules, encounters are still generated...

**Reply by jawalonoski** [reply]
> I just tried the same, but even when removing all other modules, encounters are still generated...

The [Encounter Module](https://github.com/synthetichealth/synthea/blob/master/src/main/java/org/mitre/synthea/modules/EncounterModule.java) cannot be removed and will generate encounters. For example, periodic wellness encounters with the primary care physician.

**Comment by jawalonoski** [comment]
> can we specify it somewhere and get only 50 encounters for that patient? Is it possible through Synthea? 

No.

> if it's not possible then how are you deciding the number of resources generated for a patient, Can we change that particular part of the code for our requirement?

The disease modules simulate the progression and treatment of disease, and collectively determine how many encounters occur for each patient. If you want to modify that, you have to modify the modules.

See https://github.com/synthetichealth/synthea/wiki/Module-Gallery and https://synthetichealth.github.io/module-builder/
